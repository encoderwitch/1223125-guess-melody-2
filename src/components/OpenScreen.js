import React from 'react';
import PropTypes from 'prop-types';

const OpenScreen = (props) => {
  const {time, errorCount, HandleClick} = props;

  return (
    <div>
      <h2>Rules of game</h2>
      <p>The rules are simple:</p>
      <ul>
        <li>Time for all questions {time} min. </li>
        <li>You can make {errorCount} errors.</li>
      </ul>
      <p>Good luck!</p>
      <button onClick={HandleClick}>Start of game</button>
    </div>
  );
};

OpenScreen.defaultProps = {
  time: 0,
  errorCount: 0,
};

OpenScreen.propTypes = {
  time: PropTypes.number,
  errorCount: PropTypes.number,
  HandleClick: PropTypes.func,
};

export default OpenScreen;
