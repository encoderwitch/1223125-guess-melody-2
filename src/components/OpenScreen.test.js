import React from 'react';
import renderer from 'react-test-renderer';
import OpenScreen from './OpenScreen';

it(`renders correctly`, () => {
  const tree = renderer
      .create(<OpenScreen/>)
      .toJSON();
  expect(tree).toMatchSnapshot();
});
