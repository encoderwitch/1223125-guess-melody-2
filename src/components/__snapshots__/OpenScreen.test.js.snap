// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2>
    Rules of game
  </h2>
  <p>
    The rules are simple:
  </p>
  <ul>
    <li>
      Time for all questions 
      0
       min. 
    </li>
    <li>
      You can make 
      0
       errors.
    </li>
  </ul>
  <p>
    Good luck!
  </p>
  <button>
    Start of game
  </button>
</div>
`;
